library/src/main/java/com/novoda/downloadmanager/RoomAppDatabase.java

```diff
@@ -1,12 +1,44 @@
 package com.novoda.downloadmanager;
 
 import android.arch.persistence.room.Database;
+import android.arch.persistence.room.Room;
 import android.arch.persistence.room.RoomDatabase;
+import android.content.Context;
+
+import java.util.ArrayList;
+import java.util.List;
 
 @Database(entities = {RoomBatch.class, RoomFile.class}, version = 1)
 abstract class RoomAppDatabase extends RoomDatabase {
 
+    private static RoomAppDatabase SINGLE_INSTANCE;
+
     abstract RoomBatchDao roomBatchDao();
 
     abstract RoomFileDao roomFileDao();
+
+    List<String> fileNames() {
```

Comment:
@joetimmins I've moved the logic that was in `V2DatabaseFiles` to here. Perhaps we should consider adding an interface to this `Database` so we don't leak `Room` everywhere in the app.

Reply:
mmm the app doesn't currently know that there is room under-the-hood because it only talks to `DownloadsPersistence`

Reply:
Yeah this is really the wrong place for this code to live. The class that extends RoomDatabase should basically just return `Dao`s, the `obtainInstance` method just prevents us having another class to act as a singleton for this one.

The idea of having a `V2DatabaseFiles` interface was precisely to stop the rest of the app having a concrete dependency on a Room class. Then, the `RoomV2DatabaseFiles` class was the Room-specific implementation of that interface. Had this been a proper app and not a library, that Room-specific implementation would have lived in a very small package and set to package-private, so as not to leak implementation details, but that isn't possible in a library.

Reply:
I also think that this shouldn't be here